osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs

```diff
@@ -129,13 +130,19 @@ protected override void CheckForResult(bool userTriggered, double timeOffset)
             if (userTriggered)
                 return;
 
-            // Ensure the tail can only activate after all previous ticks already have.
+            // Ensure the tail can only activate after all previous ticks/repeats already have.
             //
             // This covers the edge case where the lenience may allow the tail to activate before
             // the last tick, changing ordering of score/combo awarding.
-            if (DrawableSlider.NestedHitObjects.Count > 1 && !DrawableSlider.NestedHitObjects[^2].Judged)
+            var lastTick = DrawableSlider.NestedHitObjects.LastOrDefault(o => o.HitObject is SliderTick || o.HitObject is SliderRepeat);
+            if (lastTick?.Judged == false)
                 return;
 
+            // Attempt to preserve correct ordering of judgements as best we can by forcing
+            // an un-judged head to be missed when the user has clearly skipped it.
+            if (Tracking && !DrawableSlider.HeadCircle.Judged)
+                DrawableSlider.HeadCircle.MissForcefully();
```

Comment:
Actually as it turns out the miss mentioned above is not there on `master`. It is caused by _this_ code.

| master | this PR |
| :-: | :-: |
| ![osu_2023-10-24_14-11-47](https://github.com/ppy/osu/assets/20418176/b5f052f1-593a-4ead-acea-02aebb2b64d9) | ![osu_2023-10-24_14-14-12](https://github.com/ppy/osu/assets/20418176/bb4cf00c-65cc-4db5-b7fb-0885023e2044) |

Log output:

<details>
<summary>master</summary>

```
[runtime] 2023-10-24 12:30:51 [verbose]: osu.Game.Rulesets.Osu.Objects.SliderHeadCircle @ 133225 judged @ 133252 as LargeTickHit
[runtime] 2023-10-24 12:30:51 [debug]: Pressed (RightButton) handled by DrawableHitCircle+HitReceptor.
[runtime] 2023-10-24 12:30:51 [debug]: Pressed (LeftButton) handled by DrawableHitCircle+HitReceptor.
[runtime] 2023-10-24 12:30:51 [verbose]: osu.Game.Rulesets.Osu.Objects.SliderTailCircle @ 133375 judged @ 133376 as SmallTickMiss
[runtime] 2023-10-24 12:30:51 [verbose]: osu.Game.Rulesets.Osu.Objects.Slider @ 133225 judged @ 133376 as Ok
```

</details>

<details>
<summary>this PR</summary>

```
[runtime] 2023-10-24 12:28:31 [verbose]: osu.Game.Rulesets.Osu.Objects.SliderHeadCircle @ 133225 judged @ 133225.8112239248 as LargeTickMiss
[runtime] 2023-10-24 12:28:31 [debug]: Pressed (RightButton) handled by DrawableHitCircle+HitReceptor.
[runtime] 2023-10-24 12:28:31 [debug]: Pressed (LeftButton) handled by DrawableHitCircle+HitReceptor.
[runtime] 2023-10-24 12:28:31 [verbose]: osu.Game.Rulesets.Osu.Objects.SliderTailCircle @ 133375 judged @ 133375.55566974595 as SmallTickMiss
[runtime] 2023-10-24 12:28:31 [verbose]: osu.Game.Rulesets.Osu.Objects.Slider @ 133225 judged @ 133375.55566974595 as Miss
```

</details>

As it turns out the tail forces a miss of the head _way_ before the leniency period starts. A way to fix it would be:

```diff
diff --git a/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs b/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs
index 4872119b35..9e8ff6ae52 100644
--- a/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs
+++ b/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderTail.cs
@@ -138,6 +138,9 @@ protected override void CheckForResult(bool userTriggered, double timeOffset)
             if (lastTick?.Judged == false)
                 return;
 
+            if (timeOffset < SliderEventGenerator.TAIL_LENIENCY)
+                return;
+
             // Attempt to preserve correct ordering of judgements as best we can by forcing
             // an un-judged head to be missed when the user has clearly skipped it.
             if (Tracking && !DrawableSlider.HeadCircle.Judged)
@@ -145,7 +148,7 @@ protected override void CheckForResult(bool userTriggered, double timeOffset)
 
             // The player needs to have engaged in tracking at any point after the tail leniency cutoff.
             // An actual tick miss should only occur if reaching the tick itself.
-            if (timeOffset >= SliderEventGenerator.TAIL_LENIENCY && Tracking)
+            if (Tracking)
                 ApplyResult(r => r.Type = r.Judgement.MaxResult);
             else if (timeOffset > 0)
                 ApplyResult(r => r.Type = r.Judgement.MinResult);

```

The other tick objects don't have this problem, thankfully (added code is in `timeOffset >= 0` branches).

Reply:
Have applied this change, makes good sense.